test/clients/reports_test.exs

```diff
@@ -91,4 +91,87 @@ defmodule GlassFactoryApi.Clients.ReportsTest do
       end
     end
   end
+
+  describe "list_rates_and_costs_reports/3" do
+    test "returns reports of the given client id", %{bypass: bypass, config: config} do
+      request_response = GlassFactoryApi.Fixtures.Clients.Reports.list_rates_and_costs_reports()
+
+      Bypass.expect_once(bypass, fn conn ->
+        conn
+        |> Plug.Conn.put_resp_content_type("application/json")
+        |> Plug.Conn.resp(200, request_response)
+      end)
+
+      assert {
+               :ok,
+               [
+                 %RatesAndCostsReport{
+                   client_id: 1234,
+                   project_id: 12345,
+                   job_id: nil,
+                   activity_id: 123_456,
+                   user_id: 222,
+                   role_id: 1234,
+                   date: "2018-06-19",
+                   time: 8,
+                   rate: 35,
+                   cost: 35
+                 }
+               ]
+             } = Reports.list_rates_and_costs_reports(2079, [], config)
+    end
+
+    test "returns empty array when no rates and costs reports are found", %{
+      bypass: bypass,
+      config: config
+    } do
+      Bypass.expect_once(bypass, fn conn ->
+        Plug.Conn.resp(conn, 404, "")
+      end)
+
+      time_reports = Reports.list_rates_and_costs_reports(1, [], config)
+
+      assert time_reports == {:error, "Can't find rates and costs reports for client id 1"}
+    end
+  end
+
+  describe "list_rates_and_costs_reports!/3" do
+    test "returns rates and costs reports of the given client id", %{
+      bypass: bypass,
+      config: config
+    } do
+      request_response = GlassFactoryApi.Fixtures.Clients.Reports.list_rates_and_costs_reports()
```

Comment:
The word `response` kind of implies we already have an associated request, so prepending `request_` doesn't add much value. Maybe `expected_response` will better reveal our intention here.